### ISSUE TITLE
No longer detecting networks after updating the driver

### ISSUE THREAD

**Issue by i486** [aircrack-ng/rtl8812au#1073]
I recently installed the latest driver commit (b3227c731ec957b706f8eb2ebcf42ba1c2f0d610) for my USB-AC56 (RTL8812AU) WiFi card on my Kali virtual machine. However, after installing this driver, my WiFi card stopped detecting any networks in both monitor and managed modes. Additionally, the "Available Networks" list in the Network Manager appeared greyed out and inaccessible.

To resolve this issue, I randomly chose an earlier commit (ccda185731471e385d4b2c3a7910cb967447483d) to rollback to, and fortunately, my WiFi card started working again.

I then reinstalled the latest driver, and the same problem occurred once more.

Throughout the installation and testing process, I followed these steps: Removed the USB card, "sudo make dkms_remove," checked the DKMS tree, restarted the system, performed "make dkms_install, restarted the system, plugged the USB card " No errors were encountered during these steps.

Based on my observations, it seems like something may be incorrect in the commits between April and May.

I plan to roll back through the commits, one by one, starting from the latest version. I will update this thread later with my findings.

Various System/Wi-Fi information (when latest driver is installed and scan doesn't find anything)
[wireless-info.txt](https://github.com/aircrack-ng/rtl8812au/files/11584079/wireless-info.txt)

**Comment by i486** [comment]
I have just installed the previous version ([commit 35308f4dd73e77fa572c48867cce737449dd8548](https://github.com/aircrack-ng/rtl8812au/commit/35308f4dd73e77fa572c48867cce737449dd8548)) and now my WiFi card is working again.

[**Some cleanup of unused chips**](https://github.com/aircrack-ng/rtl8812au/commit/b3227c731ec957b706f8eb2ebcf42ba1c2f0d610). 

Well, well, well, it seems the cleanup crew had a little too much enthusiasm and accidentally swept away the vital chips that powered our WiFi. Fear not, intrepid troubleshooter! It's time to bring back those wayward chips and restore WiFi harmony to our digital realm :P

**Comment by kimocoder** [comment]
Yeah, I got aware of it. I will revert when I get home 👍

**Comment by fedepell** [comment]
Yes please, seems to be same here with a  "TP-Link Archer T2U PLUS [RTL8821AU]".

Thanks in advance!

**Comment by kimocoder** [comment]
Reverting this evening.

**Comment by vishytk** [comment]
I am using Fedora 30, Kernel-6.3.4 and the issue still persists. The solution suggested below works but the main branch still has problem.

My device is TP-Link 802.11ac

> I have just installed the previous version ([commit 35308f4dd73e77fa572c48867cce737449dd8548](https://github.com/aircrack-ng/rtl8812au/commit/35308f4dd73e77fa572c48867cce737449dd8548)) and now my WiFi card is working again.
> 
> [**Some cleanup of unused chips**](https://github.com/aircrack-ng/rtl8812au/commit/b3227c731ec957b706f8eb2ebcf42ba1c2f0d610).
> 
> Well, well, well, it seems the cleanup crew had a little too much enthusiasm and accidentally swept away the vital chips that powered our WiFi. Fear not, intrepid troubleshooter! It's time to bring back those wayward chips and restore WiFi harmony to our digital realm :P

**Comment by Entropy512** [comment]
Looking at the commit history, it looks like @kimocoder didn't get around to the revert, or is spending some more time doing a root cause analysis on why this broke things.

I've confirmed reverting that commit does fix monitor mode on an Alfa AWUS036ACH

**Comment by NayamAmarshe** [comment]
Same here. Reverting to https://github.com/aircrack-ng/rtl8812au/commit/ccda185731471e385d4b2c3a7910cb967447483d fixed it for TP-Link Archer T2U Plus.

**Comment by n0tank3sh** [comment]
@Entropy512  well the root cause of this that types are mapped according to the array before the cleanup. After the cleanup the type for rtl8821 driver is 4 while update require it to be 2. Either way I reverted that change(check my pr) or I can add padding to fix this or @kimocoder  have to make a lot of changes.

**Comment by zoogies** [comment]
Bumping this, reverting to ccda185 fixed for the TP-Link Archer T2U Plus.

**Comment by etlam262** [comment]
> Reverting this evening.

Hey @kimocoder just a small reminder, that this hasn't been fixed yet.
5ae69b685b54af6797dd7473917c22de81ded338 is working for me.

**Comment by git4rker** [comment]
waiting patiently lol
